consumer/src/main/kotlin/streams/kafka/KafkaEventSink.kt

```diff
@@ -52,54 +55,56 @@ class KafkaEventSink(private val config: StreamsConfig,
+                    KafkaManualCommitEventConsumer(kafkaConfig, log)
                 }
             }
         }
     }
 
-    private fun getErrorService(): ErrorService {
-        if (!this::errorService.isInitialized) {
-            val kafkaConfig = KafkaSinkConfiguration.create(config, db.databaseName())
-            this.errorService = KafkaErrorService(kafkaConfig.asProperties(),
-                    ErrorService.ErrorConfig.from(kafkaConfig.streamsSinkConfiguration.errorConfig),
-                    { s, e -> log.error(s,e as Throwable) })
-        }
-        return this.errorService
-    }
-
-    override fun start() { // TODO move to the abstract class
-        if (StreamsPluginStatus.RUNNING == status()) {
+    override fun start() = runBlocking { // TODO move to the abstract class
+        val topics = streamsTopicService.getTopics()
+        val isWriteableInstance = Neo4jUtils.isWriteableInstance(db)
+        if (streamsConfig.clusterOnly && !Neo4jUtils.isCluster(db)) {
             if (log.isDebugEnabled) {
-                log.debug("Kafka Sink is already started.")
+                log.info("""
+                        |Cannot init the Kafka Sink module as is forced to work only in a cluster env, 
+                        |please check the value of `${StreamsConfig.CLUSTER_ONLY}`
+                    """.trimMargin())
             }
-            return
+            return@runBlocking
         }
-        val streamsConfig = StreamsSinkConfiguration.from(config, db.databaseName())
-        val topics = streamsTopicService.getTopics()
-        val isWriteableInstance = Neo4jUtils.isWriteableInstance(db)
         if (!streamsConfig.enabled) {
             if (topics.isNotEmpty() && isWriteableInstance) {
                 log.warn("You configured the following topics: $topics, in order to make the Sink work please set ${StreamsConfig.SINK_ENABLED}=true")
             }
             log.info("The Kafka Sink is disabled")
-            return
+            return@runBlocking
         }
         log.info("Starting the Kafka Sink")
-        this.eventConsumer = getEventConsumerFactory()
-                .createStreamsEventConsumer(config, log)
-                .withTopics(topics) as KafkaEventConsumer
-        this.eventConsumer.start()
-        this.job = createJob()
+        mutex.withLock {
+            if (StreamsPluginStatus.RUNNING == status()) {
+                if (log.isDebugEnabled) {
+                    log.debug("Kafka Sink is already started.")
+                }
+                return@runBlocking
+            }
+            try {
+                eventConsumer = getEventConsumerFactory()
+                        .createStreamsEventConsumer(config, log)
+                        .withTopics(topics) as KafkaEventConsumer
+                job = createJob()
+            } catch (e: Exception) {
+                log.error("Cannot create job because of the following exception:", e)
```

Comment:
be more descriptive with this error:  what job?  What does it do?  What's the consequence?  If this error comes up, user needs to know what the consequence is.  (Probably that sink isn't working)

Reply:
done